walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -36,16 +36,34 @@ public int compare(Event a, Event b) {
       };
 
   /**
-   * Returns a list of time periods in which the meeting, specified by request, could happen.
+   * Returns a list of time periods in which the meeting, specified by request, could happen. If one
+   * or more time slots exists so that both mandatory and optional attendees can attend, it returns
+   * those time slots. Otherwise, it returns the time slots that fit just the mandatory attendees.
    *
    * @param eventsCollection the events we know about
    * @param request information about the meeting, including attendees, optional attendees, and how
    *     long it needs to be
    */
   public Collection<TimeRange> query(Collection<Event> eventsCollection, MeetingRequest request) {
```

Comment:
Not sure if you named this param, or the original file contained it, but "collection" (as in "eventsCollection") is a needlessly verbose appendage in a name. If all sequential collections had to have "collection" in the name, things would become unwieldy fast. In that case we might as well be writing [enterprise java](https://github.com/EnterpriseQualityCoding/FizzBuzzEnterpriseEdition).

Use the plural form of the noun to convey that it contains a sequential collection. In the case of maps, usually there's a better name that conveys the key-value relationship.

Reply:
Ack